library/platform_util.c

```diff
@@ -144,15 +144,15 @@ int mbedtls_platform_memmove( void *dst, const void *src, size_t num )
     return MBEDTLS_ERR_PLATFORM_ALLOC_FAILED;
 }
 
-int mbedtls_platform_memcmp( const void *buf1, const void *buf2, size_t num )
+int mbedtls_platform_memequal( const void *buf1, const void *buf2, size_t num )
```

Comment:
We're starting at at most uint32_t max even for environments where size_t might be uint64_t. If the library is accepting size_t as a buffer size (making it 32 vs 64-agnostic) I would rather suggest that the API should use size_t where possible.

Reply:
Good suggestion, but I don't think it's that significant. This would be important if the buffer size were to be around 4GB, and then many other features would fail also. memcmp() used by standard library has the same API for example. I think it is impossible to avoid all overflow errors.